src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedFieldsWalker.cs

```diff
@@ -123,6 +123,108 @@ private Symbol GetSymbolForLocation(Symbol symbol)
                 : topLevelMethod;
         }
 
+        public override BoundNode VisitCall(BoundCall node)
+        {
+            var result = base.VisitCall(node);
+            var method = node.Method;
+            if (method.IsStatic || node.ReceiverOpt is BoundThisReference)
+            {
+                ApplyMemberPostConditions(method.ContainingType,
+                    method.NotNullMembers, method.NotNullWhenTrueMembers, method.NotNullWhenFalseMembers);
+            }
+            return result;
+        }
+
+        public override BoundNode VisitPropertyAccess(BoundPropertyAccess node)
+        {
+            var result = base.VisitPropertyAccess(node);
+            var property = node.PropertySymbol;
+            if (property.IsStatic || node.ReceiverOpt is BoundThisReference)
+            {
+                var accessor = property.GetMethod;
+
+                ApplyMemberPostConditions(property.ContainingType,
+                    accessor.NotNullMembers, accessor.NotNullWhenTrueMembers, accessor.NotNullWhenFalseMembers);
+            }
+
+            return result;
+        }
+
+        protected override void PropertySetter(BoundExpression node, BoundExpression receiver, MethodSymbol setter, BoundExpression value = null)
+        {
+            base.PropertySetter(node, receiver, setter, value);
+
+            if (receiver is null || receiver is BoundThisReference)
+            {
+                ApplyMemberPostConditions(setter.ContainingType, setter.NotNullMembers, notNullWhenTrueMembers: default, notNullWhenFalseMembers: default);
+            }
+        }
+
+        private void ApplyMemberPostConditions(
+            TypeSymbol containingType,
+            ImmutableArray<string> notNullMembers,
+            ImmutableArray<string> notNullWhenTrueMembers,
+            ImmutableArray<string> notNullWhenFalseMembers)
+        {
+            if (notNullWhenTrueMembers.IsDefaultOrEmpty && notNullWhenFalseMembers.IsDefaultOrEmpty)
+            {
+                applyMemberPostConditions(notNullMembers, ref State);
```

Comment:
>applyMemberPostConditions [](start = 16, length = 25)

I think this needs to be done even when there are some of those members?

Reply:
I think this point hasn't been addressed.

---
In reply to: [387996409](https://github.com/dotnet/roslyn/pull/41675#discussion_r387996409) [](ancestors = 387996409)